dali/base/dadiags.cpp

```diff
@@ -122,7 +123,7 @@ class CDaliDiagnosticsServer: public IDaliServer, public Thread
                     mb.append(getReceiveQueueDetails(buf).str());
                 }
                 else if (0 == stricmp(id, "locks")) {
-                    mb.append(querySDS().getLocks(buf).str());
+                    querySDS().getLocks(mb);
```

Comment:
This looks like it's returning back a serialized format.. the other end is expecting plain text.
i.e. same as others in the other else if branches here.
No special handling on client side for string coming back.